cime_config/machines/Depends.perlmutter.gnu.cmake

```diff
@@ -0,0 +1,14 @@
+list(APPEND NO_INLINE_ARG_PACKING_LIST
+  eam/src/dynamics/se/inidat.F90
+)
+
+if (CMAKE_Fortran_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+
+  if (NOT DEBUG)
+    # new in gnu10, inline arg packing was causing INF values with SMS_P4x1.ne4pg2_ne4pg2.F-MMFXX
+    foreach(ITEM IN LISTS NO_INLINE_ARG_PACKING_LIST)
+      e3sm_add_flags("${ITEM}" " -fno-inline-arg-packing")
+    endforeach()
+  endif()
+
+endif()
```

Comment:
I think it'd be better to move this to `Depends.gnu[gpu].cmake` and remove several duplicated mach-specific files.

Reply:
At the time, I wasn't sure if this was something specific to the machine or compiler install. Are you seeing this helps on other machines?

Reply:
Yes, on Crusher with gcc/11.2.0: doing this in commit https://github.com/E3SM-Project/E3SM/pull/5132/commits/b25627187cc3bc6718c3ec99449af4cb071959d4 as part of #5132

Reply:
ok i made an issue https://github.com/E3SM-Project/E3SM/issues/5136